EventFilter/EcalRawToDigi/plugins/UnpackGPU.cu

```diff
@@ -272,32 +272,53 @@ namespace ecal {
         // get the next channel coordinates
         uint32_t nchannels = (block_length - 1) / 3;
 
+        bool bad_block = false;
+        __shared__ uint32_t ch_with_bad_block;
         // 1 threads per channel in this block
         for (uint32_t ich = 0; ich < nchannels; ich += NTHREADS) {
           auto const i_to_access = ich + threadIdx.x;
-          // threads outside of the range -> leave the loop
-          if (i_to_access >= nchannels)
-            break;
+          if (i_to_access == 0) {
+            ch_with_bad_block = 999999;
+          }
```

Comment:
I think in principle there should be a `__syncthreads()` here, to guarantee that all threads see the initial assignment of `ch_with_bad_block` before the `atomicMin(&ch_with_bad_block, i_to_access);` at line 314.

Reply:
I could put a `__syncthreads()` just before the `if (bad_block) {` in line 313. I think there is no need for it to be in the loop.

Reply:
My mistake. It had to be just after the initialisation.